libfwupdplugin/fu-msgpack.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#define G_LOG_DOMAIN "FuMsgpack"
+
+#include "config.h"
+
+#include "fu-msgpack-item-private.h"
+#include "fu-msgpack.h"
+
+/**
+ * fu_msgpack_new:
+ * @buf: data blob
+ * @error: (nullable): optional return location for an error
+ *
+ * Parses a buffer into messagepack items.
+ *
+ * Returns: (transfer container) (element-type FuMsgpackItem): items, or %NULL on error
+ *
+ * Since: 2.0.0
+ **/
+GPtrArray *
+fu_msgpack_parse(GByteArray *buf, GError **error)
+{
+	gsize offset = 0;
+	g_autoptr(GPtrArray) items = g_ptr_array_new_with_free_func((GDestroyNotify)g_object_unref);
+
+	g_return_val_if_fail(buf != NULL, NULL);
+	g_return_val_if_fail(error == NULL || *error == NULL, NULL);
+
+	while (offset < buf->len) {
+		g_autoptr(FuMsgpackItem) item = NULL;
+		item = fu_msgpack_item_parse(buf, &offset, error);
+		if (item == NULL)
+			return NULL;
```

Comment:
not sure how common this is going to be; but if you think it's more than once it happens it's probably worth prefixing the error to indicate which message from the buffer failed to make it easier to debug in the future.

Feel free to ignore!

Reply:
Not going to ignore, it's a good suggestion. Including the offset would make debugging much easier.  Fixed.